.github/workflows/main.yml

```diff
@@ -108,4 +108,7 @@ jobs:
           ref: refs/tags/${{ steps.poetry-plugin-export-version.outputs.version }}
 
       - name: Run pytest (poetry-plugin-export)
-        run: poetry run python -m pytest -p no:sugar -q poetry-plugin-export/tests/
+        run: |
+          # patch test mock for removed method
```

Comment:
I'd really rather not merge this with the hack present -- is there a way around it?

Reply:
Unfortunately not anything trivial. We test against the test suite of the installed version of the plugin. So, we either have to redo the fix so that create_factory is still the config provide, and wait for a forwards compatible plugin release. 

That said, I don't think the hack is too bad. Especially considering the alternative and since it's not patching anything other than the test mock.

Reply:
This is what a candidate fix might look like for this. Let me know what you think.

```diff
diff --git a/src/poetry/factory.py b/src/poetry/factory.py
index 3614324b..5d35d000 100644
--- a/src/poetry/factory.py
+++ b/src/poetry/factory.py
@@ -3,6 +3,7 @@ from __future__ import annotations
 import contextlib
 import logging
 import re
+import warnings
 
 from typing import TYPE_CHECKING
 from typing import Any
@@ -62,7 +63,8 @@ class Factory(BaseFactory):  # type: ignore[misc]
         )
 
         # Loading global configuration
-        config = Config.create()
+        with warnings.filterwarnings("ignore", category=DeprecationWarning):
+            config = self.create_config()
 
         # Loading local configuration
         local_config_file = TOMLFile(base_poetry.file.parent / "poetry.toml")
@@ -111,6 +113,16 @@ class Factory(BaseFactory):  # type: ignore[misc]
     def get_package(cls, name: str, version: str) -> ProjectPackage:
         return ProjectPackage(name, version, version)
 
+    @classmethod
+    def create_config(cls, io: IO | None = None) -> Config:
+        if io is not None:
+            logger.debug("Ignoring provided io when creating config.")
+        warnings.warn(
+            "Use of Factory.create_config() is deprecated, use Config.create() instead",
+            DeprecationWarning,
+        )
+        return Config.create()
+
     @classmethod
     def configure_sources(
         cls,
```

Reply:
This looks a lot more palatable to me. Maybe throw a `TODO` comment onto it if you think that would make it easier to remember to remove it -- otherwise that LGMT.

Reply:
Cool I will apply it then